src/main/java/carpet/script/CarpetExpression.java

```diff
@@ -2060,24 +2060,24 @@ else if ("!spectating".equalsIgnoreCase(playerName))
             return (_c, _t ) -> retval;
         });
 
-        this.expr.addLazyFunction("entity_area", 7, (c, t, lv) -> {
-            BlockPos center = new BlockPos(
-                    NumericValue.asNumber(lv.get(1).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(2).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(3).evalValue(c)).getDouble()
-            );
-            Box area = new Box(center).expand(
-                    NumericValue.asNumber(lv.get(4).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(5).evalValue(c)).getDouble(),
-                    NumericValue.asNumber(lv.get(6).evalValue(c)).getDouble()
+        this.expr.addLazyFunction("entity_area", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            BlockArgument centerLocator = BlockArgument.findIn(cc, lv, 1);
+            BlockPos startpos = centerLocator.block.getPos();
+            BlockPos corner = BlockArgument.findIn(cc, lv, centerLocator.offset).block.getPos();
+
+            Box area = new Box(startpos).expand(
+                    corner.getX(),
```

Comment:
problem is that if the second argument is an actual block argument, you will end up stretching the selected area by absolute coordinates of the other block. Like you have done in the other PRs (scan)  - if the second block argument is a world-localized block -  you should take a difference, othewise one ca assume its a relative coord

Reply:
![image](https://user-images.githubusercontent.com/53367549/88074271-7902e100-cb77-11ea-8fb3-9289a054c22b.png)
I say it makes more sense like that

Reply:
Plus, how do I tell if someone is entering a block position or if they're entering a triple of the distances?

Reply:
refer to https://github.com/gnembon/fabric-carpet/blob/5ed55ecf9e07bdcb98b803a4c845771f68c3c08f/src/main/java/carpet/script/CarpetExpression.java#L2176

Reply:
ok thanks